renderer/components/UI/Select.js

```diff
@@ -70,40 +70,45 @@ const StyledInput = styled(BasicInput)`
 
 const itemToString = item => (item ? item.value : '')
 
+const getInitialSelectedItem = (items, initialSelectedItem) => {
+  return initialSelectedItem
+    ? {
+        initialInputValue: itemToString(initialSelectedItem),
+        initialSelectedItem: items.find(i => i.key === initialSelectedItem),
+      }
+    : {}
+}
+
 /**
- * @render react
- * @name Select
+ * Internationalizes select items
+ *
+ * @param {Array} items
+ * @callback messageMapper key=>intl message mapper
+ * @param {intlShape} intl
+ * @returns {Array}
  */
-class Select extends React.PureComponent {
-  static displayName = 'Select'
-
-  static propTypes = {
-    color: PropTypes.string,
-    fieldApi: PropTypes.object.isRequired,
-    fieldState: PropTypes.object.isRequired,
-    iconSize: PropTypes.number,
-    initialSelectedItem: PropTypes.string,
-    intl: intlShape.isRequired,
-    items: PropTypes.array,
-    onValueSelected: PropTypes.func,
-    theme: PropTypes.object.isRequired,
-  }
-
-  static defaultProps = {
-    items: [],
-    iconSize: 8,
-  }
-
-  inputRef = React.createRef()
-
-  blurInput = () => {
-    if (this.inputRef.current) {
-      this.inputRef.current.blur()
+function useIntl(items, messageMapper, intl) {
+  return useMemo(() => {
+    // if messageMapper is not set just original items array
+    // in this case items should already contain value prop for each key
+    if (!messageMapper) {
+      return items
     }
-  }
+    return items.map(({ key }) => {
+      return { key, value: intl.formatMessage({ ...messageMapper(key) }) }
```

Comment:
It might be worth adding a check here to verify that `.messageMapper(key)` returns something and if not just return the item key as the value.

Reply:
This has pros and cons. Currently, mapper works as a "compile time" checker. so if the message is missing it will throw an error which would be easily detectable. Silently falling back to a key is ok, but may bring more problems than improvements. What do you think?